File: python_solutions/transpose_matrix.py
from typing import List

class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:

        rows = len(matrix[0])
        columns=len(matrix)

        transposed = [[0 for _ in range(columns)] for _ in range(rows)]

        for i in range(rows):
            for j in range(columns):
                transposed[i][j]=matrix[j][i]
       
        return transposed
    
if __name__ == "__main__":
    solution = Solution()
    initial = [
        [1, 2, 3], 
        [4, 5, 6]
    ]
    print(solution.transpose(initial))

File: javaSolutions/src/com/practice/TransposeMatrix.java
package com.practice;

import java.util.Arrays;

public class TransposeMatrix {

    public static void main(String[] args) {
        int[][] initial = {
                {1, 2, 3},
                {4, 5, 6}
        };

        int[][] solution = transpose(initial);
        System.out.println(Arrays.deepToString(solution));
    }

    public static int[][] transpose(int[][] matrix) {
        int rows = matrix.length;
        int cols = matrix[0].length;

        int[][] transposed = new int[cols][rows];

        for (int r = 0; r < rows; r++) {
            for (int c = 0; c < cols; c++) {
                transposed[c][r] = matrix[r][c];
            }
        }

        return transposed;
    }
}